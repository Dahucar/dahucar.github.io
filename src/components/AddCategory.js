import React, { useState } from 'react';
import PropTypes from 'prop-types'

// el metodo enviado desde otro componente se recive en los props
const AddCategory = ({ setCategories }) => {
  const [ inputValue, setInputValue ] = useState('');
  const handleChangeInputValue = ( e ) => {
    setInputValue( e.target.value );
  }

  const handleSubmitForm = ( e ) => {
    console.log('handleSubmitForm', inputValue);
    e.preventDefault();
    // a qui se llama al metodo que se recive desde los props en el por referencia se tendra el listado de categorias.

    if (inputValue.trim().length >= 2) {
      setCategories( categ =>  [ inputValue, ...categ ] );
      setInputValue( '' );
    }

  } 

  return (
    <form onSubmit={ handleSubmitForm }>
      <input 
        type="text"
        value={ inputValue }
        placeholder="Ingrese un nombre"
        onChange={ handleChangeInputValue }
      />
      <span>{ inputValue }</span>
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}

export default AddCategory;