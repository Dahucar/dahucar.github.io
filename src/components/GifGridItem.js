import React from 'react'

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card 
                    animate__animated 
                    animate__bounce
                    animate__fadeIn">
      <h4>{ title }</h4>
      <img src={ url } alt={ title } />
    </div>
  )
}

