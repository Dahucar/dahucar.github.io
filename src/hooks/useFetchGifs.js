// Mi custom hook es esta funcion que serÃ¡ exportada
import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";

// esta puede usar otros hooks sin problemas 
export const useFetchGifs = ( category ) => {

  const [customHookState, setCustomHookState] = useState({
    data: [],
    loading: true
  });

  useEffect(() => {
    getGifs( category ).then( img => {
      setCustomHookState({
        data: img,
        loading: false
      });
    });
  }, [ category ]);

  return customHookState; // es la informacion que se le pasa al 'useState'

}
