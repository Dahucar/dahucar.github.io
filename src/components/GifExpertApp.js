import React, { useState } from 'react';
import AddCategory from './AddCategory';
import { GifGrid } from './GifGrid';

const GitExpertApp = () => {
  // el listado de categorias se enviara por refencia a los compoenentes que se deseen
  const [categories, setCategories] = useState([ 'Halo' ]);
  return (
    <>
      <h2>Gif Expert App</h2>
      {/* aqui se estan enviado por referencia las categorias y le metodo para guardar una nueva */}
      <AddCategory setCategories={ setCategories } />
      <hr />
      <ol>
        {
          categories.map( category => <GifGrid key={ category } category={ category } />)
        }
      </ol>
    </>
  )
}

export default GitExpertApp;