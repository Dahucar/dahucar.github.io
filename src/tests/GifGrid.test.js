import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../components/GifGrid';
import { useFetchGifs } from '../hooks/useFetchGifs';
// 1. The jest.mock function allows me to pretend I have real information from my custom hook
jest.mock('../hooks/useFetchGifs');

describe('Testing on GifGrid component', () => {
  
  const category = 'ASUS';
  test('Should generate snapshot of GifGrid component', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: false
    });
    const wrapper = shallow( <GifGrid category={ category } /> );
    expect( wrapper ).toMatchSnapshot();
  });

  test('Should be displayed the gif items when the images is loading', () => {
    // 2. Here, i created the data for using in my custom hook
    const gifs = [{
      id: 'ABC01',
      title: 'Test img 01',
      url: 'https://test/url/ABC01'
    },
    {
      id: 'ABC02',
      title: 'Test img 02',
      url: 'https://test/url/ABC02'
    }];

    // 3. With the mockReturnValue i can update the informacion
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow( <GifGrid category={ category } /> );
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('p').exists() ).toBe( false );
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
  });
  
});
