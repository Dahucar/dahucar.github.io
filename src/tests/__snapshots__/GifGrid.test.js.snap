// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on GifGrid component Should be displayed the gif items when the images is loading 1`] = `
<Fragment>
  <h3>
     
    ASUS
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC01"
      key="ABC01"
      title="Test img 01"
      url="https://test/url/ABC01"
    />
    <GifGridItem
      id="ABC02"
      key="ABC02"
      title="Test img 02"
      url="https://test/url/ABC02"
    />
  </div>
</Fragment>
`;

exports[`Testing on GifGrid component Should generate snapshot of GifGrid component 1`] = `
<Fragment>
  <h3>
     
    ASUS
     
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
